### PR TITLE
[LoopInterchange] Don't rely on ASSERTS build for tests. NFC.

### PR DESCRIPTION
**Pull request by sjoerdmeijer**
A lot of interchange tests unnecessary relied on a build with ASSERTS enabled. Instead, simply check the IR output for both negative and positive tests so that we don't rely on debug messages. This increases test coverage as these tests will now also run  with non-assert builds. For a couple of files keeping some of the debug tests was useful, so separated out them out and moved them to a similarly named *-remarks.ll file.

### CODE REVIEW

**Comment by sjoerdmeijer** [comment]
I will precommit this change soon, adding you as reviewers as a FYI, as I think this is just a NFC reshuffle of tests that increases test coverage; and no (fundamental) changes have been made to the tests itself, just the CHECKing.

File: llvm/test/Transforms/LoopInterchange/interchange-no-deps.ll [hunk 1]
```diff
@@ -1,22 +1,42 @@
-; REQUIRES: asserts
-; RUN: opt < %s -passes='loop(loop-interchange),simplifycfg' -cache-line-size=64 -simplifycfg-require-and-preserve-domtree=1 -pass-remarks-output=%t \
+; NOTE: Assertions have been autogenerated by utils/update_test_checks.py UTC_ARGS: --version 5
+; RUN: opt < %s -passes='loop(loop-interchange),simplifycfg' -simplifycfg-require-and-preserve-domtree=1 \
```
Comment (fhahn) [inline llvm/test/Transforms/LoopInterchange/interchange-no-deps.ll:2]: Why does this drop the check lines for the remarks? Those don't require asserts, only `-stats` does I think, which is still here?

Comment (fhahn) [reply llvm/test/Transforms/LoopInterchange/interchange-no-deps.ll:2]: Also, why drop `-cache-line-size=64`?

File: llvm/test/Transforms/LoopInterchange/call-instructions-remarks.ll [hunk 2]
```diff
@@ -0,0 +1,63 @@
+; REQUIRES: asserts
+; RUN: opt < %s -passes=loop-interchange -pass-remarks-missed='loop-interchange' -pass-remarks-output=%t -S \
```
Comment (fhahn) [inline llvm/test/Transforms/LoopInterchange/call-instructions-remarks.ll:2]: Remarks don't require asserts I think

File: llvm/test/Transforms/LoopInterchange/interchange-no-deps.ll [hunk 3]
```diff
@@ -29,23 +49,42 @@ for1.inc:
...
+; STATS-SAME: ptr [[PTR:%.*]]) {
+; STATS-NEXT:  [[ENTRY:.*]]:
+; STATS-NEXT:    br label %[[FOR1_HEADER:.*]]
+; STATS:       [[FOR1_HEADER]]:
+; STATS-NEXT:    [[INDVARS_IV19:%.*]] = phi i64 [ 0, %[[ENTRY]] ], [ [[INDVARS_IV_NEXT20:%.*]], %[[FOR1_INC:.*]] ]
+; STATS-NEXT:    br label %[[FOR2:.*]]
+; STATS:       [[FOR2]]:
+; STATS-NEXT:    [[INDVARS_IV:%.*]] = phi i64 [ 0, %[[FOR1_HEADER]] ], [ [[INDVARS_IV_NEXT:%.*]], %[[FOR2]] ]
+; STATS-NEXT:    store i64 [[INDVARS_IV]], ptr [[PTR]], align 4
+; STATS-NEXT:    [[INDVARS_IV_NEXT]] = add nuw nsw i64 [[INDVARS_IV]], 1
```
Comment (fhahn) [inline llvm/test/Transforms/LoopInterchange/interchange-no-deps.ll:70]: Should be `CHECK-`?

**Review by fhahn — COMMENTED** [review]
Remarks shouldn't require asserts and checking the remark if interchange didn't happen seems like a more straight-forward way then checking the IR in some cases

**Comment by sjoerdmeijer** [comment]
Thanks for the comments, I will follow up soon to:
- fix the `STATS` vs. `CHECK` tag mismatch,
- restore some of the remarks.

I dropped the `cache-line-size` for some tests because I verified it didn't make a difference.

**Pull request merged**